operators/src/engine/mod.rs

```diff
@@ -77,3 +77,46 @@ macro_rules! call_on_generic_raster_processor {
         }
     };
 }
+
+/// Calls a function on two `TypedRQueryProcessor`s by calling it on their variant combination.
+/// Call via `call_bi_generic_processor!(input_a, input_b, (processor_a, processor_b) => function)`.
+#[macro_export]
+macro_rules! call_bi_generic_processor {
```

Comment:
i'am still not conviced that this will scale up for more then 3 inputs. We should really discuss why type conversion as adapter step for n >=4 inputs is such a problem if we do projection, scaling and other expensive operations anyway.

Reply:
we should discuss this independently of this PR

Reply:
ok